src/oic/oic/provider.py

```diff
@@ -2049,3 +1883,449 @@ def get_by_sub_and_(self, sub: str, key: str, val: Any) -> Optional[str]:
+        """
+        if cookie:
+            headers = cookie
+        else:
+            headers = []
+
+        self.sdb.set_verify_logout(uid)
+
+        if redirect_uri is not None:
+            redirect = redirect_uri
+        else:
+            redirect = "/"
+        try:
+            tmp_id_token_hint = esr["id_token_hint"]
+        except KeyError:
+            tmp_id_token_hint = ""
+
+        context = {
+            "id_token_hint": tmp_id_token_hint,
+            "post_logout_redirect_uri": esr["post_logout_redirect_uri"],
+            "key": self.sdb.get_verify_logout(uid),
+            "redirect": redirect,
+            "action": "/" + EndSessionEndpoint("").etype,
+        }
+        return Response(
+            self.template_renderer("verify_logout", context), headers=headers
+        )
+
+    def _get_uid_from_cookie(
+        self, cookie: Union[str, SimpleCookie, None]
+    ) -> Tuple[Optional[CookieDealer], Optional[str], Optional[str]]:
+        """
+        Get cookie_dealer, client_id and uid from cookie.
+
+        :param cookie: Received cookie
+        :return: Tuple containing CookieDealer instance, client ID and User ID
+        """
+        if cookie is None:
+            return None, None, None
+
+        cookie_dealer = CookieDealer(srv=self)
+        client_id = uid = None
+
+        _cval = cookie_dealer.get_cookie_value(cookie, self.sso_cookie_name)
+        if _cval:
+            (value, _ts, typ) = _cval
+            if typ == "sso":
+                uid, client_id = value.split(DELIM)
+
+        return cookie_dealer, client_id, uid
+
+    def do_back_channel_logout(
+        self, cinfo: dict, sub: str, sid: str
+    ) -> Optional[Tuple[str, str]]:
+        """
+        Prepare information to be used to do a back-channel logout.
+
+        :param cinfo: Client information
+        :param sub: Subject identifier
+        :param sid: The Issuer ID
+        :return: Tuple with logout URI and signed logout token
+        """
+        try:
+            back_channel_logout_uri = cinfo["backchannel_logout_uri"]
+        except KeyError:
+            return None
+
+        # always include sub and sid so I don't check for
+        # backchannel_logout_session_required
+
+        payload = {
+            "sub": sub,
+            "sid": sid,
+            "events": {BACK_CHANNEL_LOGOUT_EVENT: {}},
+            "jti": uuid.uuid4().hex,
+        }
+
+        try:
+            alg = cinfo["id_token_signed_response_alg"]
+        except KeyError:
+            alg = self.capabilities["id_token_signing_alg_values_supported"][0]
+
+        _jws = JWT(self.keyjar, iss=self.name, lifetime=86400, sign_alg=alg)
+        sjwt = _jws.pack(aud=cinfo["client_id"], **payload)
+
+        return back_channel_logout_uri, sjwt
+
+    def clean_sessions(self, usids: List[str]):
+        """
+        Remove Session IDs from the session DB.
+
+        :param usids: List of session IDs
+        """
+        _sdb = self.sdb
```

Comment:
This assignment seem unnecessary...

Reply:
It makes things go a bit faster but if you don't care about that.

Reply:
That would probably depend on implementation, wouldn't it?

It is probably fine either way, was just curious.